tests/unittest/test_services.py

```diff
@@ -1479,10 +1482,10 @@ def test_k8s_registry_create(self, mock_kube_interface):
         )
 
         mock_kube_interface.create.assert_any_call(
-            KubernetesResourceType.ROLEBINDING,
-            f"{name3}-role-binding",
+            KubernetesResourceType.SECRET_GENERIC,
+            "spark8t-sa-conf-" + name3,
             namespace=namespace3,
-            **{"role": f"{name3}-role", "serviceaccount": sa3_obj.id},
+            **{"from-env-file": ANY},
         )
 
```

Comment:
I'm _really_ confused why these changes were necessary for a green CI. This unittest is for `KubeInterface`, for which -- according to my understanding -- no execution flow changes were supposed to have impacted this workflow... Strange...

Reply:
I really don't know. This sounds very strange. Can we get to the bottom of this?

Reply:
Sure, I can provide like debugging video or so going through the "before" and "after" workflows with a debugger.

Reply:
Just for the record, this was followed up offline. @deusebio suggested to add the following change https://github.com/canonical/spark-k8s-toolkit-py/pull/20 which indeed is satisfactory for `assert_any_call()`, and the highly misleading error (referring to call mismatch on `SECRET GENERIC`) is gone once corresponding parameters for `SERVICEACCOUNT` and `ROLEBINDING` are inline.

(My bad, I'm trying to avoid mocked-called-with tests in general -- however I'd note that this one is a nice one here verifying key points of the account creation, that we "insisnt on" to have.)